excuse-maker-cra/src/components/SelectPage.js

```diff
@@ -1,67 +1,105 @@
-import React,{ useState, useEffect, useCallback } from "react";
-import {Link, useParams} from "react-router-dom";
+import React, { useState, useEffect } from "react";
+import { Link, useParams } from "react-router-dom";
 import data from "../db/data.json";
-import { Row, Col} from 'antd';
-import { Button } from 'antd';
-import{RedoOutlined,CopyOutlined,SwapLeftOutlined} from '@ant-design/icons'
+import { Row, Col } from "antd";
+import { Button } from "antd";
+import {
+  RedoOutlined,
+  CopyOutlined,
+  SwapLeftOutlined,
+} from "@ant-design/icons";
 
-function colStyle({span,offset,value}){
-    return(
-        <Col span={span} offset={offset} value={value}>
-            <div style={{display:"flex",flexDirection: "column",alignItems: "center"}}/>
-        </Col>
-    );
+function ColStyle({ span, offset, value, children }) {
+  return (
+    <Col span={span} offset={offset} value={value}>
+      <div
+        style={{
+          display: "flex",
+          flexDirection: "column",
+          alignItems: "center",
+        }}
+      />
+      {children}
+    </Col>
+  );
 }
 
-function SelectPage(){
-    const {selected} = useParams();
-    console.log(selected);
-    const [excuseText,setExcuseText] = useState();
+function randomValueExcuseArray(list) {
+  const random = Math.floor(Math.random() * 10);
 
-    useEffect(()=>{
-        setExcuseText(randomValueExcuseArray(selected));
-    },[]);
-    
-    function randomValueExcuseArray(selected){
-        const random = Math.floor(Math.random() * 10);
-        
-        return data[selected][random];
+  console.log("랜덤", random);
+  return list[random];
+}
+
+function SelectPage() {
+  const { selected } = useParams();
+  const [excuseText, setExcuseText] = useState();
+  const [data, setData] = useState();
+
+  const UseFetch = async (url) => {
+    const response = await fetch(url);
+    const responseJson = await response.json();
+
+    console.log(url);
+    console.log(responseJson);
+    return responseJson;
+  };
+
+  useEffect(async () => {
+    const list = await UseFetch(`http://localhost:4000/${selected}`);
```

Comment:
```suggestion
    const list = await UseFetch(`http://localhost:4000/${selected}`);
```

url 의 상수화를 하면 좋을 것 같다.

```
const API_URL = 'http://localhost:4000'

const EXCUSE_LIST = (selected) =>`${API_URL}/${selected}`

const list = await UseFetch(EXCUSE_LIST(selected));
```